### PR TITLE
samples: cellular: lwm2m_client: add nRF9161DK support

### PR DESCRIPTION
**Pull request by juhaylinen**
Add configuration files for 9161DK.
Update documentation.

### CODE REVIEW

**Review by SeppoTakalo — APPROVED** [review]

File: samples/cellular/lwm2m_client/provisioning.rst [hunk 1]
```diff
@@ -25,7 +25,7 @@ Programming the AT Client sample
 
 You must program the :ref:`at_client_sample` sample to your device to control the security tags in the modem.
 See `nRF91 AT Commands Reference Guide <AT Commands Reference Guide_>`_ for documentation on each AT command.
-Also, you must provision the bootstrap credentials for the security tag (that you have specified in :kconfig:option:`CONFIG_LWM2M_CLIENT_UTILS_BOOTSTRAP_TLS_TAG` Kconfig option) to the nRF9160 modem.
+Also, you must provision the bootstrap credentials for the security tag (that you have specified in :kconfig:option:`CONFIG_LWM2M_CLIENT_UTILS_BOOTSTRAP_TLS_TAG` Kconfig option) to the nRF91 Series modem.
```
Comment (divipillai) [inline samples/cellular/lwm2m_client/provisioning.rst:28]:
```suggestion
Also, you must provision the bootstrap credentials for the security tag (that you have specified in :kconfig:option:`CONFIG_LWM2M_CLIENT_UTILS_BOOTSTRAP_TLS_TAG` Kconfig option) to the cellular modem.
```

File: samples/cellular/lwm2m_client/provisioning.rst [hunk 2]
```diff
@@ -101,5 +101,5 @@ To configure and program the sample, complete the following steps:
 1. Make the sample programmable to multiple devices by removing all hard coded credentials. This can be done by setting the :ref:`CONFIG_APP_LWM2M_PSK <CONFIG_APP_LWM2M_PSK>` Kconfig option to empty value.
 #. Enable bootstrapping using the configuration overlay file :file:`overlay-avsystem-bootstrap.conf`.
    Bootstrapping is required for an LwM2M client to rotate security credentials.
-#. Prepare the production script or steps for your nRF9160-based device.
+#. Prepare the production script or steps for your nRF91-based device.
```
Comment (divipillai) [inline samples/cellular/lwm2m_client/provisioning.rst:104]:
```suggestion
#. Prepare the production script or steps for your nRF91 Series device.
```

**Review by divipillai — COMMENTED** [review]
For "Evaluating LwM2M Advanced Firmware Update", most of the commands are related nRF9160 and related modem firmware (https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/cellular/lwm2m_client/fota.html#fota-upload-command). Do we plan to have similar one for nRF9161 also?

**Comment by divipillai** [comment]
Please note we have a change log entry in https://github.com/nrfconnect/sdk-nrf/pull/12062, in general, for cellular samples.
So, we won't require a separate change log entry for this PR. 
cc: @eivindj-nordic

**Comment by juhaylinen** [comment]
> For "Evaluating LwM2M Advanced Firmware Update", most of the commands are related nRF9160 and related modem firmware (https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/cellular/lwm2m_client/fota.html#fota-upload-command). Do we plan to have similar one for nRF9161 also?

I'll  update the fota doc soon.

**Comment by juhaylinen** [comment]
@divipillai I've updated the fota doc. Please check.

File: samples/cellular/lwm2m_client/fota.rst [hunk 3]
```diff
@@ -142,21 +143,35 @@ The following example output shows the allocation of a resource id ``lwm2m_clien
...
    [INFO] fota.py - Upload app_update.bin with instance 0 to Coiote
-   [INFO] fota.py - Binary app_update.bin, Size 336032 (bytes)
+   [INFO] fota.py - Binary app_update.bin, Size 364747 (bytes)
    [INFO] coiote.py - Creating fota resource for binary app_update.bin with id lwm2m_client_fota_instance_0
    [INFO] fota.py - Allocated Resource id lwm2m_client_fota_instance_0 for instance 0
 
-.. code-block:: console
+.. tabs::
+
+   .. group-tab:: nRF9160 DK
```
[the remark this answers is not in the document]
Comment (divipillai) [reply samples/cellular/lwm2m_client/fota.rst:152]: Please interchange the tabs. This is the format we follow across the docs. Applies for all the cases.

File: samples/cellular/lwm2m_client/fota.rst [hunk 4]
```diff
@@ -142,21 +143,35 @@ The following example output shows the allocation of a resource id ``lwm2m_clien
...
+
+      .. code-block:: console
+
+         ./scripts/fota.py upload 1 mfw_nrf9160_update_from_1.x.x_to_1.x.x-FOTA-TEST.bin
+
+      where 1.x.x is the latest modem release version.
```
Comment (divipillai) [inline samples/cellular/lwm2m_client/fota.rst:158]:
```suggestion
      where 2.x.x is the latest modem release version.
```

File: samples/cellular/lwm2m_client/fota.rst [hunk 5]
```diff
@@ -142,21 +143,35 @@ The following example output shows the allocation of a resource id ``lwm2m_clien
...
+   .. group-tab:: nRF9161 DK
+
+      .. code-block:: console
+
+         ./scripts/fota.py upload 1 mfw_nrf91x1_update_from_2.x.x_to_2.x.x-FOTA-TEST.bin
 
-   ./scripts/fota.py upload 1 mfw_nrf9160_update_from_1.3.4_to_1.3.4-FOTA-TEST.bin
+      where 2.x.x is the latest modem release version.
 
 The following example output shows the allocation of a resource id ``lwm2m_client_fota_instance_0`` on the server side:
```
Comment (divipillai) [inline samples/cellular/lwm2m_client/fota.rst:168]:
```suggestion
The following is an example output that shows the allocation of a resource id ``lwm2m_client_fota_instance_0`` on the server side for the nRF9160 DK:
```

File: samples/cellular/lwm2m_client/fota.rst [hunk 6]
```diff
@@ -176,9 +191,23 @@ The update default task timeout is 800 seconds.
 
 Following is an example of updating a modem instance by giving a binary file:
 
-.. code-block:: console
+.. tabs::
 
-   ./scripts/fota.py -id urn:imei:351358811331351 update 1 file mfw_nrf9160_update_from_1.3.4_to_1.3.4-FOTA-TEST.bin
+   .. group-tab:: nRF9160 DK
```
Comment (divipillai) [inline samples/cellular/lwm2m_client/fota.rst:196]: same as above

File: samples/cellular/lwm2m_client/fota.rst [hunk 7]
```diff
@@ -264,23 +328,33 @@ Complete the following steps to test the advanced FOTA firmware update with the
...
+               ./scripts/fota.py -id urn:imei:351358811331351 update 1 file mfw_nrf9160_update_from_1.x.x-FOTA-TEST_to_1.x.x.bin
+
+         .. group-tab:: nRF9161 DK
+
+            .. code-block:: console
 
-         ./scripts/fota.py -id urn:imei:351358811331351 update 1 file mfw_nrf9160_update_from_1.x.x-FOTA-TEST_to_1.x.x.bin
+               ./scripts/fota.py -id urn:imei:359746166785274 update 1 mfw_nrf91x1_update_from_2.x.x-FOTA-TEST_to_2.x.x.bin
 
       Following is an example output of the command:
```
Comment (divipillai) [inline samples/cellular/lwm2m_client/fota.rst:351]:
```suggestion
      Following is an example output of the command for an nRF9160 DK:
```

**Review by divipillai — APPROVED** [review]

**Review by jheiskan81 — APPROVED** [review]

**Pull request merged**